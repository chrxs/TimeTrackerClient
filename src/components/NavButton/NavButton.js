import React from 'react'
import PropTypes from 'prop-types'
import { NavLink } from 'react-router-dom'

import styles from './NavButton.scss'

const NavButton = ({
  to,
  exact,
  label,
  onClick
}) => {
  if (to) {
    return (
      <NavLink
        to={to}
        className={styles.NavButton}
        activeClassName={styles.NavButtonSelected}
        exact={exact}
      >
        <div className={styles.icon} />
        <div className={styles.label}>{ label }</div>
      </NavLink>
    )
  }
  return (
    <button
      type='button'
      className={styles.NavButton}
      onClick={onClick}
    >
      <div className={styles.icon} />
      <div className={styles.label}>{ label }</div>
    </button>
  )
}

NavButton.displayName = 'NavButton'

NavButton.propTypes = {
  to: PropTypes.string,
  exact: PropTypes.bool,
  label: PropTypes.string,
  onClick: PropTypes.func
}

NavButton.defaultProps = {
  label: '',
  exact: true,
  onClick: () => {}
}

export default NavButton
