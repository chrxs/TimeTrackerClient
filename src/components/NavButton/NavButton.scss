.NavButton {
  background: none;
  border: none;
  cursor: pointer;
  display: flex;
  flex-direction: column;
  align-items: center;
  text-decoration: none;
  width: 60px;
  color: #000;
}

.NavButtonSelected {
  color: red;
}

.icon {
  border: solid 2px #ccc;
  border-radius: 100%;
  box-sizing: border-box;
  height: 50px;
  width: 50px;
}

.label {
  margin: 6px 0 0;
  text-align: center;
  font-size: 12px;
  white-space: nowrap;
}
