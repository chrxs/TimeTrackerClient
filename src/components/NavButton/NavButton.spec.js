/* global describe, test, expect */
import React from 'react'
import { BrowserRouter } from 'react-router-dom'

import NavButton from './NavButton'
import renderer from 'react-test-renderer'

describe('NavButton snapshots', () => {
  test('with "to" prop', () => {
    const component = renderer.create(
      <BrowserRouter>
        <NavButton to='/somewhere' label='going somewhere' />
      </BrowserRouter>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('without "to" prop', () => {
    const component = renderer.create(<NavButton onClick={() => {}} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
