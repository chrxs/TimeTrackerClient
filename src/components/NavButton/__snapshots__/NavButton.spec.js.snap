// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavButton snapshots with "to" prop 1`] = `
<a
  className="NavButton"
  href="/somewhere"
  onClick={[Function]}
  style={undefined}
>
  <div
    className="icon"
  />
  <div
    className="label"
  >
    going somewhere
  </div>
</a>
`;

exports[`NavButton snapshots without "to" prop 1`] = `
<button
  className="NavButton"
  onClick={[Function]}
  type="button"
>
  <div
    className="icon"
  />
  <div
    className="label"
  >
    
  </div>
</button>
`;
